tests/functional/test_contract_event.py

```diff
@@ -281,3 +281,9 @@ def test_contract_decode_logs_no_abi(owner, contract_instance):
     assert events[0].event_name == "NumberChange"
     assert events[0].newNum == 1
     assert events[0].transaction_index == 0
+
+
+def test_contract_log_container(owner, contract_instance):
+    receipt = contract_instance.setNumber(1, sender=owner)
+    assert "NumberChange" in receipt.events
+    assert receipt.events["NumberChange"]["newNum"] == 1
```

Comment:
My idea was something more akin to this:
```suggestion
    # at least 1 NumberChange event with arg `newNum` set to 1
    assert contract_instance.NumberChange(newNum=1) in receipt.events
```
Allowing more strict comparisons like this:
```py
    # exactly 1 event (NumberChange) with arg `newNum` set to 1
    assert receipt.events == [contract_instance.NumberChange(newNum=1)]
```

Reply:
but if i have lots of attributes in the event and i dont care all about them so actually dont want to create the object but only care lets say 1 attribute then i am stuck

Reply:
you could also do this:

```py
    # exactly 1 event (NumberChange), don't care what `newNum` is set to
    assert receipt.events == [contract_instance.NumberChange()]
```

Reply:
hmm, I do like this approach!

except call it `.filter` instead of `.find`

if it still returns a list we can leverage it's truthiness via:

```py
assert receipt.events.filter(contract.NumberChange, oldNum=1, newNum=2)
```

or if you want an exact check:

```py
assert len(receipt.events.filter(contract.NumberChange, oldNum=1, newNum=2)) == 1
```

Reply:
> it requires one to specify all the event attributes in order for the event to match, even in cases where you only care about event type

Just responding to this, the intent was to have un-specified event args work as wildcards. It would be a bit of work to get a proper wildcard system working

Reply:
the current implementation should be ok